marker/builders/line.py

```diff
@@ -222,14 +219,16 @@ def get_all_lines(self, document: Document, provider: PdfProvider):
                     )
                 )
 
-                # If fixing lines, mark every line to be passed to the OCR model
-                for provider_line in merged_provider_lines:
-                    provider_line.line.text_extraction_method = (
-                        "hybrid" if self.format_lines else "pdftext"
-                    )
-                page_lines[document_page.page_id] = (
-                    merged_provider_lines + detected_only_lines
-                )
+                if detected_only_lines:
+                    # If not all the lines are captured, then make sure we OCR the page
+                    document_page.text_extraction_method = "surya"
+                    boxes_to_ocr[document_page.page_id].extend(detection_boxes)
```

Comment:
Minor - Why not just return and extend by the `detected_only_lines`. The non-missed lines might actually be fine.

Reply:
I think we should tradeoff some performance for accuracy - in this case, it is possible the PDF is missing text for all lines